frontend/demo/src/reducers/data/conversations/index.ts

```diff
@@ -0,0 +1,156 @@
+import {ActionType, getType} from 'typesafe-actions';
+import {combineReducers} from 'redux';
+import {cloneDeep} from 'lodash-es';
+
+import {ResponseMetadata} from '../../../model/ResponseMetadata';
+import {Conversation} from '../../../model/Conversation';
+import * as actions from '../../../actions/conversations';
+import {Message} from '../../../model/Message';
+
+type Action = ActionType<typeof actions>;
+
+type MergedConversation = Conversation & {
+  blocked?: boolean;
+  metadata?: ResponseMetadata & {
+    loading: boolean;
+  };
+};
+
+export type AllConversationMetadata = ResponseMetadata & {
+  loading?: boolean;
+  loaded?: boolean;
+  filteredTotal?: number;
+  badgeUnreadCount?: number;
+};
+
+export type ConversationMap = {
+  [conversationId: string]: MergedConversation;
+};
+
+export type AllConversationsState = {
+  items: ConversationMap;
+  metadata: AllConversationMetadata;
+};
+
+export type ErrorState = {
+  [conversationId: string]: string;
+};
+
+export type ConversationsState = {
+  all: AllConversationsState;
+  errors: ErrorState;
+};
+
+function mergeConversations(
+  oldConversation: {[conversation_id: string]: MergedConversation},
+  newConversations: MergedConversation[]
+): ConversationMap {
+  newConversations.forEach((conversation: MergedConversation) => {
+    if (conversation.contact) {
```

Comment:
This can be collapsed to `if(conversation.contact && !conversation.contact.displayName)`